aztk/spark/helpers/submit.py

```diff
@@ -13,79 +15,11 @@ def __get_node(spark_client, node_id: str, cluster_id: str) -> batch_models.Comp
 
 
-def __app_submit_cmd(
-        spark_client,
-        cluster_id: str,
-        name: str,
-        app: str,
-        app_args: str,
-        main_class: str,
-        jars: List[str],
-        py_files: List[str],
-        files: List[str],
-        driver_java_options: str,
-        driver_library_path: str,
-        driver_class_path: str,
-        driver_memory: str,
-        executor_memory: str,
-        driver_cores: str,
-        executor_cores: str):
-    cluster = spark_client.get_cluster(cluster_id)
-    master_id = cluster.master_node_id
-    master_ip = __get_node(spark_client, master_id, cluster_id).ip_address
-
-    spark_home = constants.DOCKER_SPARK_HOME
-
-    # set file paths to correct path on container
-    files_path = '/batch/workitems/{0}/{1}/{2}/wd/'.format(cluster_id, "job-1", name)
-    jars = [files_path + jar for jar in jars]
-    py_files = [files_path + py_file for py_file in py_files]
-    files = [files_path + f for f in files]
-
-    # 2>&1 redirect stdout and stderr to be in the same file
-    spark_submit_cmd = CommandBuilder(
-        '{0}/bin/spark-submit'.format(spark_home))
-    spark_submit_cmd.add_option(
-        '--master', 'spark://{0}:7077'.format(master_ip))
-    spark_submit_cmd.add_option('--name', name)
-    spark_submit_cmd.add_option('--class', main_class)
-    spark_submit_cmd.add_option('--jars', jars and ','.join(jars))
-    spark_submit_cmd.add_option('--py-files', py_files and ','.join(py_files))
-    spark_submit_cmd.add_option('--files', files and ','.join(files))
-    spark_submit_cmd.add_option('--driver-java-options', driver_java_options)
-    spark_submit_cmd.add_option('--driver-library-path', driver_library_path)
-    spark_submit_cmd.add_option('--driver-class-path', driver_class_path)
-    spark_submit_cmd.add_option('--driver-memory', driver_memory)
-    spark_submit_cmd.add_option('--executor-memory', executor_memory)
-    spark_submit_cmd.add_option('--driver-cores', driver_cores)
-    spark_submit_cmd.add_option('--executor-cores', executor_cores)
-
-    spark_submit_cmd.add_argument(
-        '/batch/workitems/{0}/{1}/{2}/wd/'.format(cluster_id, "job-1", name) +
-        app + ' ' + ' '.join(['\'' + app_arg + '\'' for app_arg in app_args if app_args]))
-
-    if cluster.gpu_enabled:
-        docker_exec_cmd = CommandBuilder('sudo nvidia-docker exec')
-    else:
-        docker_exec_cmd = CommandBuilder('sudo docker exec')
-    
-    docker_exec_cmd.add_option('-i', constants.DOCKER_SPARK_CONTAINER_NAME)
-    docker_exec_cmd.add_argument('/bin/bash  >> {0} 2>&1 -c \"cd '.format(
-        constants.SPARK_SUBMIT_LOGS_FILE) + files_path + '; ' + spark_submit_cmd.to_str() + '\"')
-
-    return [
-        docker_exec_cmd.to_str()
-    ]
-
-
-def submit_application(spark_client, cluster_id, application, wait: bool = False):
-    """
-    Submit a spark app
-    """
-
+def generate_task(spark_client, container_id, application):
```

Comment:
this looks very similar to the generate_task method in aztk/spark/helpers/job_submission.py. Can we only write it once?

Reply:
I think that unfortunately, they are sufficiently different to require two different methods. 

This one takes aztk applications, uploads the necessary resource files and creates Batch task objects from them. 

The generate_task in job_submission.py takes an array of aztk application definitions, serializes the definitions and uploads the serialized object as a resource file for a the job manager task it creates.